collate.go

```diff
@@ -77,21 +85,24 @@ func collationType(value interface{}) token {
 	if value == nil {
 		return kNull
 	}
-	switch value := value.(type) {
-	case bool:
-		if !value {
+
+	v := reflect.ValueOf(value)
+	switch v.Kind() {
+	case reflect.Bool:
+		if !v.Bool() {
 			return kFalse
 		}
 		return kTrue
-	case float64, uint64, uint16, json.Number:
+	case reflect.Float64, reflect.Uint64, reflect.Uint16: //json.Number?
```

Comment:
I don't think we can just ignore json.Number, can we?  Have you done any testing to see what json.Number returns when passed to reflect.ValueOf()?

Reply:
json.Number is reported as "string" type by reflect, I have added a specific type assertion for json.Number is reflect returns a string type.